### PR TITLE
[Mellanox] Allow flexible setting of I2C transactions

### PR DESCRIPTION
**Pull request by nazariig**
Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

### CODE REVIEW

**Comment by nazariig** [comment]
@lguohan please have a look

File: patch/0014-mlxsw-i2c-Allow-flexible-setting-of-I2C-transactions.patch [hunk 1]
```diff
@@ -0,0 +1,409 @@
+From 1b96f4deebde7974332d9c498a559bb0931675c4 Mon Sep 17 00:00:00 2001
+From: Vadim Pasternak <vadimp@mellanox.com>
+Date: Wed, 22 Jul 2020 20:26:51 +0300
+Subject: [PATCH net:backport] mlxsw: i2c: Allow flexible setting of I2C
+ transactions size
+
+Current implementation uses fixed size of I2C data transaction buffer.
```
Comment (lguohan) [inline patch/0014-mlxsw-i2c-Allow-flexible-setting-of-I2C-transactions.patch:7]: what is the upstream commit id?

Comment (lguohan) [reply patch/0014-mlxsw-i2c-Allow-flexible-setting-of-I2C-transactions.patch:7]: can you identify the upstream commit id for this patch?

Comment (nazariig) [reply patch/0014-mlxsw-i2c-Allow-flexible-setting-of-I2C-transactions.patch:7]: @lguohan commit details:
```
commit 3029a693beda5bff599feccc3beb9df1e141c786
Author: Vadim Pasternak <vadimp@mellanox.com>
Date:   Wed May 29 11:47:16 2019 +0300

    mlxsw: i2c: Allow flexible setting of I2C transactions size
    
    Current implementation uses fixed size of I2C data transaction buffer.
    Allow to set size of I2C transactions according to I2C physical adapter
    capability. For that purpose adapter read and write size is obtained
    from the I2C physical adapter and buffer size is set according to the
    minimum of these two values. If adapter does not provide such info,
    default buffer size is to be used.
    It allows to improve performance of I2C access to silicon when long
    size transactions are used.
    
    Signed-off-by: Vadim Pasternak <vadimp@mellanox.com>
    Acked-by: Jiri Pirko <jiri@mellanox.com>
    Signed-off-by: Ido Schimmel <idosch@mellanox.com>
    Signed-off-by: David S. Miller <davem@davemloft.net> 
```

Comment (lguohan) [reply patch/0014-mlxsw-i2c-Allow-flexible-setting-of-I2C-transactions.patch:7]: ok. can you add this original commit in the patch description?

Comment (nazariig) [reply patch/0014-mlxsw-i2c-Allow-flexible-setting-of-I2C-transactions.patch:7]: @lguohan i'll check with the maintainers and let you you know.

**Comment by liat-grozovik** [comment]
@lguohan please ignore this PR. let use a newer one with the comments requested.

**Comment by lguohan** [comment]
close upon request.

**Pull request closed without merging**